content/src/content/jcr_root/apps/core/wcm/components/modal/v1/modal/clientlibs/site/js/modal-lib.js

```diff
@@ -0,0 +1,26 @@
+/*******************************************************************************
+ * Copyright 2019 Adobe Systems Incorporated
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+(function(tingle, root) {
+    "use strict";
+
+    function ModalLib(opts) {
+        return new root.tingle.modal(opts);
```

Comment:
I would rather we don't just blindly return the Tingle.js object. Instead, I think we should return our custom object and provide our own options and methods (such as `open` and `setContent`) so that we can setup a proper API skeleton.